### PR TITLE
Speedup fasta export

### PR DESCRIPTION
**Pull request by cmdcolin**
I found several places for optimizing FASTA exports with regards to #854. With the pythium data (340 genes), the time taken is now down to about 25s with this PR (repeated runs 28s,23.7s,22.7s)

Therefore, this new method is about ~16x faster

Some notes

1) This stops using SequenceChunks table. SequenceChunks does not really do anything for us compared with just reading the sequence from file. As a comparison, if I add back SequenceChunks, then the times are ~5x longer than without it (takes 1m59s,1m52s). Still better than 6min30s for export, but again, we don't really need to query database for those chunks.

2)  The getResiduesFromFeature had support for multiple feature locations, but it did something weird where if one of the feature locations was on the negative strand, then all of the residues are flipped. That assumption seemed unnecessary, so I just made it check strand for each chunk. Multiple feature locations are currently unused in our codebase, but in any case, having this model is probably more correct

The new FASTA outputs are bit-identical to 2.0.1 according to file hashes too

### CODE REVIEW

File: grails-app/services/org/bbop/apollo/SequenceService.groovy [hunk 1]
```diff
@@ -39,22 +39,19 @@ class SequenceService {
      * @param feature
      * @return
      */
-    String getResiduesFromFeature(Feature feature ) {
-        List<FeatureLocation> featureLocationList = FeatureLocation.createCriteria().list {
-            eq("feature",feature)
-            order("fmin","asc")
+    String getResiduesFromFeature(Feature feature) {
+        String returnResidues = ""
+        def orderedFeatureLocations = feature.featureLocations.sort { it.fmin }
```
Comment (nathandunn) [inline grails-app/services/org/bbop/apollo/SequenceService.groovy]: nice, doing it memory vs a separate sql query will be much faster, esp. if already there via a join

Comment (cmdcolin) [inline grails-app/services/org/bbop/apollo/SequenceService.groovy]: Yep, I used this thing called "javaflamegraph" to identify those pain points https://www.npmjs.com/package/javaflamegraph

Pretty easy to setup, just found it today too...

Comment (nathandunn) [inline grails-app/services/org/bbop/apollo/SequenceService.groovy]: that is interesting . . . will take a look

**Comment by nathandunn** [comment]
Interesting, but not suprising that the chunk code wasn't doing anything helpful.  This was adopted from what Ed had done originally in-memory, but is obviously not a pinch-point here.

**Comment by nathandunn** [comment]
for 50 genes: 15 seconds -> 1.7 seconds . . .  I tested both FASTA (all types) and GFF + FASTA . . . . looked good to me

**Pull request merged**